src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs

```diff
@@ -244,135 +225,131 @@ Language ID
-
+        private static string GetLocalizedKeyboardLayoutName(string layoutDisplayName)
+        {
+            if (layoutDisplayName != null && Environment.OSVersion.Version.Major >= 5)
+            {
                 StringBuilder sb = new StringBuilder(512);
-
                 uint res = UnsafeNativeMethods.SHLoadIndirectString(layoutDisplayName, sb, (uint)sb.Capacity, IntPtr.Zero);
-
-                if (res == NativeMethods.S_OK) {
+                if (res == NativeMethods.S_OK)
+                {
                     return sb.ToString();
                 }
             }
 
             return null;
         }
-        
-        /// <include file='doc\InputLanguage.uex' path='docs/doc[@for="InputLanguage.CreateInputLanguageChangedEventArgs"]/*' />
+
         /// <devdoc>
-        ///     Creates an InputLanguageChangedEventArgs given a windows message.
+        /// Creates an InputLanguageChangedEventArgs given a windows message.
         /// </devdoc>
-        /// <internalonly/>
-        internal static InputLanguageChangedEventArgs CreateInputLanguageChangedEventArgs(Message m) {
+        internal static InputLanguageChangedEventArgs CreateInputLanguageChangedEventArgs(Message m)
+        {
             return new InputLanguageChangedEventArgs(new InputLanguage(m.LParam), unchecked((byte)(long)m.WParam));
         }
 
-        /// <include file='doc\InputLanguage.uex' path='docs/doc[@for="InputLanguage.CreateInputLanguageChangingEventArgs"]/*' />
         /// <devdoc>
-        ///     Creates an InputLanguageChangingEventArgs given a windows message.
+        /// Creates an InputLanguageChangingEventArgs given a windows message.
         /// </devdoc>
-        /// <internalonly/>
-        internal static InputLanguageChangingEventArgs CreateInputLanguageChangingEventArgs(Message m) {
+        internal static InputLanguageChangingEventArgs CreateInputLanguageChangingEventArgs(Message m)
+        {
             InputLanguage inputLanguage = new InputLanguage(m.LParam);
 
             // NOTE: by default we should allow any locale switch
-            //
-            bool localeSupportedBySystem = !(m.WParam == IntPtr.Zero);
+            bool localeSupportedBySystem = m.WParam != IntPtr.Zero;
             return new InputLanguageChangingEventArgs(inputLanguage, localeSupportedBySystem);
-
         }
 
-        /// <include file='doc\InputLanguage.uex' path='docs/doc[@for="InputLanguage.Equals"]/*' />
         /// <devdoc>
-        ///    <para>Specifies whether two input languages are equal.</para>
+        /// Specifies whether two input languages are equal.
         /// </devdoc>
-        public override bool Equals(object value) {
-            if (value is InputLanguage) {
-                return(this.handle == ((InputLanguage)value).handle);
-            }
-            return false;
+        public override bool Equals(object value)
+        {
+            return value is InputLanguage other && this.handle == other.handle;
```

Comment:
nit: can you parenthesize both of these checks for clarity

```suggestion
            return (value is InputLanguage other) && (this.handle == other.handle);
```